docs/development/AzureFunctions.md

```diff
@@ -1,10 +1,40 @@
 ﻿# Azure Functions Integration
 
+This document describes how dd-trace-dotnet integrates with Azure Functions for distributed tracing.
+
+**Related Documentation:**
+- [Azure Functions Architecture Deep Dive](AzureFunctions-Architecture.md) - Detailed architectural information about Azure Functions Host and .NET Worker
+
 Azure functions operates in one of two ways:
 
 - In-process
 - Isolated (out of process)
 
+## NuGet Package Architecture
+
+### Datadog.AzureFunctions Package
+
+The `Datadog.AzureFunctions` NuGet package is the primary instrumentation package for Azure Functions. It contains:
+
+- **Datadog.Trace.dll** - The full managed tracer with all auto-instrumentation code
+- **Native profiler binaries** - Platform-specific profiler libraries (e.g., `Datadog.Trace.ClrProfiler.Native.so` for Linux)
+- **Monitoring home directory structure** - Deployed to `/home/site/wwwroot/datadog/` (Linux) or equivalent on Windows
+
+When profiling is enabled globally (via `CORECLR_ENABLE_PROFILING=1`), the native profiler loads `Datadog.Trace.dll` into **both the host process (`func.exe` or `Microsoft.Azure.WebJobs.Script.WebHost`) and the worker process**. This allows instrumentation code to run in both processes, which is essential for:
+- Instrumenting host process methods (e.g., `GrpcMessageConversionExtensions.ToRpcHttp()`)
+- Instrumenting worker process methods (e.g., `FunctionExecutionMiddleware`)
+- Ensuring correct trace context propagation across the process boundary
+
+### Datadog.Trace Package
+
+The `Datadog.Trace` NuGet package is different from `Datadog.AzureFunctions`:
+
+- **Contains only** `Datadog.Trace.Manual.dll` - The manual instrumentation API
+- **Does NOT contain** auto-instrumentation code or native profiler binaries
+- Used by application code for manual instrumentation (e.g., `Tracer.Instance.StartActive()`)
+
+In Azure Functions scenarios, the worker application references `Datadog.Trace` for manual instrumentation, while `Datadog.AzureFunctions` provides the auto-instrumentation. The version of `Datadog.Trace` referenced by the application doesn't need to match `Datadog.AzureFunctions` exactly, as the auto-instrumentation comes entirely from `Datadog.Trace.dll` bundled in `Datadog.AzureFunctions`.
```

Comment:
> The version of `Datadog.Trace` referenced by the application

This seems confusing, an application just shouldn't reference Datadog.Trace directly (because it's pulled in by _Datadog.AzureFunctions_?

Reply:
Reworded in #7799